tests/validation/tests/v3_api/test_multi_cluster_app.py

```diff
@@ -201,3 +617,74 @@ def validate_app_version(project_client, multiclusterapp, app_id):
         project_client.list_app(name=app_id).data[0].externalId
     assert mcapp_template_version == app_template_version, \
         "App Id is different from the Multi cluster app id"
+
+
+def return_application_status_and_upgrade(client1, app_id1, client2, app_id2):
+    app_data1 = client1.list_app(id=app_id1).data
+    application1 = app_data1[0]
+    app_data2 = client2.list_app(id=app_id2).data
+    application2 = app_data2[0]
+    a = application1.state == "active" \
+        and application1.answers["mysqlUser"] == "admin1234"
+    b = application2.state == "active" \
+        and application2.answers["mysqlUser"] == "admin1234"
+    return a == True and b != True
+
+
+def validate_app_upgrade_mca(multiclusterapp):
+    for i in range(0, len(multiclusterapp.targets)):
+        project_client = project_detail["p_client" + str(i)]
+        app = multiclusterapp.targets[0].projectId.split(":")
+        appid = app[1] + ":" + multiclusterapp.targets[i].appId
+        temp_version = multiclusterapp.templateVersionId
+        app = temp_version.split(":")[1].split("-")
+        mcapp_template_version = "catalog://?catalog=" + app[0] + \
+                                 "&template=" + app[1] + "&version=" \
+                                 + app[2]
+        app_template_version = \
+            project_client.list_app(id=appid).data[0].externalId
+        assert mcapp_template_version == app_template_version, \
+            "App Id is different from the Multi cluster app id"
+
+
+def validate_deletion_mca(multiclusterapp):
+    for i in range(0, len(multiclusterapp.targets)):
+        app_id = multiclusterapp.targets[i].appId
+        assert app_id is not None, "app_id is None"
+        project_client = project_detail["p_client"+str(i)]
+        app = multiclusterapp.targets[i].projectId.split(":")
+        app1id = app[1] + ":" + multiclusterapp.targets[i].appId
+        validate_app_deletion(project_client, app1id)
+
+
+def validate_all_answer_override_mca(multiclusterapp):
+    for i in range(0, len(multiclusterapp.targets)):
+        project_client = project_detail["p_client" + str(i)]
+        app = multiclusterapp.targets[0].projectId.split(":")
+        appid = app[1] + ":" + multiclusterapp.targets[i].appId
+        hold = multiclusterapp['answers'][0]
+        val = hold["values"]
+        app_answers = \
+            project_client.list_app(id=appid).data[0].answers
+        assert str(val) == str(app_answers), \
+            "App answers are different than the Multi cluster answers"
+
+
+def validate_answer_override(project_client, appid, i):
```

Comment:
Modify this method to take in actual question and answer as parameters instead of "i" which is the actual target index from the list of MCAPP targets